src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs

```diff
@@ -83,16 +91,14 @@ public object Read()
             return serializer.Deserialize(jsonReader, manifest.SagaEntityType);
         }
 
-        static void WriteWithinLock(Action<JsonWriter, IContainSagaData> action, JsonWriter writer, string targetPath, DateTime lastModificationSeenAt, IContainSagaData sagaData = null)
+        static void WriteWithinLock(Action action, string targetPath, DateTime lastModificationSeenAt)
```

Comment:
👎 To this change. The way it was before, we completely avoid having to close over any values, which should prevent the need for a per-call delegate allocation.